src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -794,6 +794,17 @@ public static SyntaxToken WithoutTrailingTrivia(this SyntaxToken token)
             return token.WithTrailingTrivia(default(SyntaxTriviaList));
         }
 
+        /// <summary>
+        /// Gets the node within the given <paramref name="root"/> corresponding to the given <paramref name="span"/>.
+        /// If the <paramref name="span"/> is <see langword="null"/>, then returns the given <paramref name="root"/> node.
+        /// </summary>
+        public static SyntaxNode GetNodeForSpan(this SyntaxNode root, TextSpan? span, bool findInTrivia, bool getInnermostNodeForTie)
```

Comment:
i feel like this is just an overload of `FindNode`.  Can we call it that, just with an optional span argument?